lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureCloudBlobClient.cs

```diff
@@ -67,14 +68,30 @@ public ICloudBlockBlob GetBlockBlobReference(Uri uri)
             return new AzureCloudBlockBlob(uri, _client.Credentials);
         }
 
-        public BlobResultSegment ListBlobsSegmented(string prefix, bool useFlatListing, BlobListingDetails blobListingDetails,
-            int? maxResults, BlobContinuationToken continuationToken, BlobRequestOptions blobRequestOptions,
+        public BlobResultSegment ListBlobsSegmented(
+            string containerName,
+            string relativeAddress,
+            bool useFlatListing,
+            BlobListingDetails blobListingDetails,
+            int? maxResults,
+            BlobContinuationToken continuationToken,
+            BlobRequestOptions blobRequestOptions,
             OperationContext operationContext)
         {
-            var task = _client.ListBlobsSegmentedAsync(prefix, useFlatListing, blobListingDetails, maxResults,  continuationToken, 
-                                                        blobRequestOptions, operationContext);
-            task.Wait();
-            return task.Result;
+            CloudBlobContainer container = _client.GetContainerReference(containerName);
+            CloudBlobDirectory directory = container.GetDirectoryReference(relativeAddress);
+            return directory.ListBlobsSegmentedAsync(
+                useFlatListing,
+                blobListingDetails,
+                maxResults,
+                continuationToken,
+                blobRequestOptions,
+                operationContext).GetAwaiter().GetResult();
+        }
+
+        public ContainerResultSegment ListContainersSegmented(BlobContinuationToken continuationToken)
+        {
+            return _client.ListContainersSegmentedAsync(continuationToken).GetAwaiter().GetResult();
```

Comment:
same as above

Reply:
Replied on the other issue and I will follow the same resolution here once that is addressed.

---
In reply to: [196655252](https://github.com/apache/reef/pull/1464#discussion_r196655252) [](ancestors = 196655252)